operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java

```diff
@@ -141,22 +116,37 @@ public UpdateControl<Keycloak> reconcile(Keycloak kc, Context<Keycloak> context)
         }
 
         if (modifiedSpec) {
-            return UpdateControl.updateResource(kc);
+            // just patch spec using SSA, nothing more
+            Keycloak patchedKc = new KeycloakBuilder()
+                    .withNewMetadata()
+                        .withName(kc.getMetadata().getName())
+                        .withNamespace(kc.getMetadata().getNamespace())
+                    .endMetadata()
+                    .withSpec(kc.getSpec())
+                    .build();
+            return UpdateControl.patchResource(patchedKc);
         }
 
-        var upgradeLogicControl = upgradeLogicFactory.create(kc, context, deploymentDependentResource, updateJobDependentResource)
+        var existingDeployment = context.getSecondaryResource(StatefulSet.class).orElse(null);
+        ContextUtils.storeOperatorConfig(context, config);
+        ContextUtils.storeWatchedResources(context, watchedResources);
+        ContextUtils.storeDistConfigurator(context, distConfigurator);
+        ContextUtils.storeCurrentStatefulSet(context, existingDeployment);
+        ContextUtils.storeDesiredStatefulSet(context, new KeycloakDeploymentDependentResource().desired(kc, context));
```

Comment:
Is it possible to inject the instance rather than creating a new one?

Reply:
I was initially considering that but decided not to do it. Didn't want to add a managed DR to CDI. Felt like mixing two distinct concepts and that it might be confusing that e.g. injections in `KeycloakDeploymentDependentResource` sometimes work and sometimes don't – based on whether the instance was created using the Operator, or CDI.

But either way works for me. Do you want to still do it?